presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -294,6 +345,278 @@ public Page getOutput()
         return null;
     }
 
+    private static class PartitionData
+    {
+        private static final int INITIAL_POSITION_COUNT = 1024;
+        private static final int DEFAULT_ELEMENT_SIZE_IN_BYTES = 8;
+
+        private final int partition;
+        private final AtomicLong rowsAdded;
+        private final AtomicLong pagesAdded;
+        private final PagesSerde serde;
+        private final Lifespan lifespan;
+
+        private int positionCount;  // number of rows for this partition
+        private int channelCount;
+        private int[] positions;   // the default positions array
+        private int[] rowSizes;
+        private BlockEncodingBuffers[] blockEncodingBuffers;
+        private int capacity;
```

Comment:
The value of `capacity` never changes. Make it final and initialize when declaring.

```
        ....
        private final PagesSerde serde;
        private final Lifespan lifespan;
        private final int capacity = DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
```